ecosystem/typescript/sdk/src/bcs/helper.ts

```diff
@@ -45,6 +45,32 @@ export function deserializeVector(deserializer: Deserializer, cls: any): any[] {
   return list;
 }
 
+// Recursive function to serialize n-dimensional arrays
+function serializeNestedArrayWithFunc(value: any[], func: string, serializer: Serializer): void {
+  if (Array.isArray(value[0])) {
```

Comment:
It is better to validate and handle the case that `value` is null/undefined as the SDK can be used in scenarios where TS type hints are unavailable.